### ISSUE TITLE
MIDIIn.connectAll connects to wrong ports on Linux

### ISSUE THREAD

**Issue by sensestage** [ModalityTeam/Modality-toolkit#196]
If the other ports on Linux are higher in the list, then SC connects to itself instead of the external program.
Proper behaviour would be to skip the own ports and connect to the next one.

**Comment by sensestage** [comment]
fix:
```
/*

MIDIClient.init;

MIDIClient.externalSources;
MIDIClient.externalDestinations;

MIDIIn.connectAll;
*/

+ MIDIClient {

	*externalSources{
		^MIDIClient.sources.select({ |src,i|
			src.device != "SuperCollider"
		})
	}

	*externalDestinations{
		^MIDIClient.destinations.select({ |src,i|
			src.device != "SuperCollider"
		})
	}

}

+ MIDIIn{

	*connectAll {
		if(MIDIClient.initialized.not,{ MIDIClient.init });
		MIDIClient.externalSources.do({ |src,i|
			MIDIIn.connect(i,src);
		});
	}
}
```

**Comment by sensestage** [comment]
See also: https://github.com/supercollider/supercollider/issues/1487

**Comment by sensestage** [comment]
Fixed in e9ff254cdbc137f2b95902c1e0782e55a47d1aac
